File: Assets/StreamingAssets/Server/skyway_models/Pad.py
class Pad:
    def __init__(self, id, node, status="available"):
        self.id = id
        self.node = node
        self.status = status

    def is_available(self):
        return self.status == "available"

    def occupy(self):
        if self.is_available():
            self.status = "occupied"

    def release(self):
        self.status = "available"

File: Assets/Scripts/skyway models/Pad.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Pad : MonoBehaviour
{
    [SerializeField]
    string id;

    [SerializeField]
    Node node;

    [SerializeField]
    bool isAvailable;

    void Awake()
    {
        id = Guid.NewGuid().ToString();
    }

    void Start()
    {
        isAvailable = true;
    }

    public string GetId()
    {
        return id;
    }

    public Node GetNode()
    {
        return node;
    }

    [SerializeField]
    public bool IsAvailable()
    {
        return isAvailable;
    }

    [SerializeField]
    void Occupy()
    {
        isAvailable = false;
    }

    [SerializeField]
    void Release()
    {
        isAvailable = true;
    }

    public SerializablePad ToSerializablePad()
    {
        return new SerializablePad()
        {
            id = this.id,
            node = this.node.GetId(),
            isAvailable = this.isAvailable,
        };
    }
}
